core/src/com/unciv/models/ruleset/Ruleset.kt

```diff
@@ -320,13 +320,7 @@ class Ruleset {
         }
 
         for (resource in tileResources.values) {
-            if (resource.revealedBy != null && !technologies.containsKey(resource.revealedBy!!))
-                lines += "${resource.name} revealed by tech ${resource.revealedBy} which does not exist!"
-            if (resource.improvement != null && !tileImprovements.containsKey(resource.improvement!!))
-                lines += "${resource.name} improved by improvement ${resource.improvement} which does not exist!"
-            for (terrain in resource.terrainsCanBeFoundOn)
-                if (!terrains.containsKey(terrain))
-                    lines += "${resource.name} can be found on terrain $terrain which does not exist!"
+            warningCount += resource.checkModLinks(this, lines)
```

Comment:
I really don't think this is good. Mod links should be checked in the ruleset, not in the resource...

Reply:
Marginal decision either way. Makes property access a bit shorter on average was the idea methinks.